File: Examples/Cross Machine Platform/CrossPlatform/CS_WPF/OkumaInterface/OkumaMill.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using DataApi = Okuma.CMDATAPI.DataAPI;
using DataEnums = Okuma.CMDATAPI.Enumerations;

namespace CS_WPF.OkumaInterface
{
    public class OkumaMill : IOkuma, IDisposable
    {

        DataApi.CMachine _machine;
        DataApi.CVariables _variables;

        public OkumaMill()
        {
            _machine = new DataApi.CMachine();
            _machine.Init();
            _variables = new DataApi.CVariables();
        }

        public OperationModeEnum GetOperationMode()
        {
            //Convert the Okuma.CMDATAPI.Enumerations.OperationMode to our local OperationModeEnum Enumeration
            return (OperationModeEnum)_machine.GetOperationMode();
        }

        public void WriteVariable(int variableNumber, double newValue)
        {
            _variables.SetCommonVariableValue(variableNumber, newValue);
        }

        public double ReadVariable(int variableNumber)
        {
            return _variables.GetCommonVariableValue(variableNumber);
        }

        public void Dispose()
        {
            if (_machine != null)
            {
                _machine.Close();
                _variables = null;
            }
        }
    }
}

File: Examples/Cross Machine Platform/VB_WPF/OkumaInterface/OkumaMill.vb
Imports Okuma.CMDATAPI.DataAPI
Public Class OkumaMill
    Implements IOkuma

    Private _machine As CMachine
    Private _variables As CVariables

    Public Sub New()
        _machine = New CMachine()
        _machine.Init()
        _variables = New CVariables()

    End Sub

    Public Function ReadVariable(variableNum As Integer) As Double Implements IOkuma.ReadVariable

        Return _variables.GetCommonVariableValue(variableNum)

    End Function

    Public Sub WriteVariable(variableNum As Integer, newValue As Double) Implements IOkuma.WriteVariable
        _variables.SetCommonVariableValue(variableNum, newValue)
    End Sub

    Public Function GetOperationMode() As OperationModeEnum Implements IOkuma.GetOperationMode
        Return CType(_machine.GetOperationMode(), OperationModeEnum)
    End Function

    Public Sub Dispose() Implements IDisposable.Dispose
        If _machine IsNot Nothing Then
            _machine.Close()
            _variables = Nothing
        End If
    End Sub
End Class
